src/site/paragraphs/table.drupal.liquid

```diff
@@ -1,57 +1,84 @@
-            "value": {
-                "0": [
-                "Row 1 Column 1",
-                "Row 1 Column 2",
-                "Row 1 Column 3"
-                ],
-                "1": [
-                "Row 2 Column 1",
-                "Row 2 Column 2",
-                "Row 2 Column 3"
-                ],
-                "2": [
-                "Row 3 Column 1",
-                "Row 3 Column 2",
-                "Row 3 Column 3"
-                ],
-            }
-            "format": "rich_text",
-            "caption": "Test Table"
-        }
-    }
+Example data:
+entity: {
+"entityId": "3944",
+"fieldTable": {
+"tableValue": "Row 1 Column 1 Row 1 Column 2 Row 1 Column 3 Row 2 Column 1 Row 2 Column 2 Row 2 Column 3 Row 3 Column 1
+Row 3 Column 2 Row 3 Column 3",
+"value": {
+"0": [
+"Row 1 Column 1",
+"Row 1 Column 2",
+"Row 1 Column 3"
+],
+"1": [
+"Row 2 Column 1",
+"Row 2 Column 2",
+"Row 2 Column 3"
+],
+"2": [
+"Row 3 Column 1",
+"Row 3 Column 2",
+"Row 3 Column 3"
+],
+}
+"format": "rich_text",
+"caption": "Test Table"
+}
+}
 {% endcomment %}
 
+<!-- Mobile -->
+<div class="vads-u-display--flex vads-u-flex-direction--column medium-screen:vads-u-display--none" data-template="paragraphs/table-mobile">
+  <!-- Assign mobile table data -->
+  {% assign mobileTableData = entity | formatMobileTableData %}
 
-<table data-template="paragraphs/table" data-entity-id="{{ entity.entityId }}" class="va-table">
+  <!-- Caption -->
+  <p><strong>{{ mobileTableData.caption }}</strong></p>
+
+  <!-- Show all -->
 
-    {% if entity.fieldTable.caption %}
-        <caption>{{ entity.fieldTable.caption }}</caption>
-    {% endif %}
 
-    {% for value in entity.fieldTable.value %}
-        {% if forloop.first == true %}
-            <thead>
-                <tr>
-                    {% for column in value %}
-                        <th>{{ column }}</th>
-                    {% endfor %}
-                </tr>
-            </thead>
-        {% elsif forloop.last == false %}
-            <tbody>
-                <tr class="class">
-                    {% for column in value %}
-                        <td>{{ column }}</td>
-                    {% endfor %}
-                </tr>
-            </tbody>
-        {% endif %}
+  <!-- Iterate over rows -->
+  {% for row in mobileTableData.rows %}
+    <!-- First key-value pair -->
+    <p class="vads-u-margin-bottom--0"><strong>{{ row.dataPoints.0.label }}</strong> </p>
+	<div class="mobile-value vads-u-display--flex .vads-u-margin-x--1px"  >{{ row.dataPoints.0.value }}</div>
+    <!-- Additional info toggle -->
+    {% for dataPoint in row.dataPoints %}
+      {% unless forloop.first %}
+        <p><strong>{{ dataPoint.label }}:</strong> {{ dataPoint.value }}</p>
+      {% endunless %}
     {% endfor %}
 
+    <!-- Line break -->
+    <hr/>
```

Comment:
I think `<hr/>` is generally used for a thematic break in language, whereas for stylizing we should try to use margin + borders.